ratis-server/src/main/java/org/apache/ratis/server/raftlog/segmented/SegmentedRaftLogWorker.java

```diff
@@ -371,21 +372,32 @@ private void flushIfNecessary() throws IOException {
         if (stateMachineDataPolicy.isSync()) {
           stateMachineDataPolicy.getFromFuture(f, () -> this + "-flushStateMachineData");
         }
-        final Timer.Context logSyncTimerContext = raftLogSyncTimer.time();
         flushBatchSize = (int)(lastWrittenIndex - flushIndex.get());
-        out.flush();
-        logSyncTimerContext.stop();
+        if (flushExecutorWorkQueue.isEmpty()) {
```

Comment:
I think adding another flush work isn't necessary if there are already flush tasks in the queue that haven't been executed yet.

Reply:
Good point!